xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/security/XarSecurityRule.java

```diff
@@ -78,4 +80,33 @@ public RuleState getState()
     {
         return RuleState.DENY;
     }
+
+    @Override
+    public int hashCode()
+    {
+        return new HashCodeBuilder(3, 17).append(right).append(simple)
+                .append(securityTool).toHashCode();
+    }
+
+    @Override
+    public boolean equals(Object object)
+    {
+        if (object == null) {
+            return false;
+        }
+        if (object == this) {
+            return true;
+        }
+        if (object.getClass() != getClass()) {
+            return false;
+        }
+
+        XarSecurityRule rhs = (XarSecurityRule) object;
+        return new EqualsBuilder()
+                .appendSuper(super.equals(object))
```

Comment:
I think this line means that for the XarSecurityRule to be equals it needs to be the same (since that's how `Object#equals()` works).

A unit test would be nice :)

Reply:
Added in  d712af2 ; also removed the `.appendSuper(…)`